src/main/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdk.java

```diff
@@ -0,0 +1,147 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk;
+
+import static com.google.cloud.tools.managedcloudsdk.OsInfo.Name.WINDOWS;
+
+import com.google.cloud.tools.managedcloudsdk.command.CommandExitException;
+import com.google.cloud.tools.managedcloudsdk.command.CommandFactory;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponentInstaller;
+import com.google.cloud.tools.managedcloudsdk.install.SdkInstaller;
+import com.google.cloud.tools.managedcloudsdk.update.SdkUpdater;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/** A manager for installing, configuring and updating the Cloud SDK. */
+public class ManagedCloudSdk {
+
+  private final Version version;
+  private final Path managedSdkDirectory;
+  private final OsInfo osInfo;
+
+  /** Instantiated with {@link ManagedCloudSdk#newManagedSdk}. */
+  ManagedCloudSdk(Version version, Path managedSdkDirectory, OsInfo osInfo) {
+    this.version = version;
+    this.managedSdkDirectory = managedSdkDirectory;
+    this.osInfo = osInfo;
+  }
+
+  public Path getSdkHome() {
+    return managedSdkDirectory.resolve(version.getVersion()).resolve("google-cloud-sdk");
+  }
+
+  /** Returns a path to gcloud executable (operating system specific). */
+  public Path getGcloud() {
+    return getSdkHome()
+        .resolve("bin")
+        .resolve(osInfo.name().equals(WINDOWS) ? "gcloud.cmd" : "gcloud");
+  }
+
+  /** Simple check to verify Cloud SDK installed by verifying the existence of gcloud. */
+  public boolean isInstalled() {
+    if (getSdkHome() == null) {
+      return false;
+    }
+    if (!Files.isDirectory(getSdkHome())) {
+      return false;
+    }
+    if (!Files.isRegularFile(getGcloud())) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Query gcloud to see if component is installed. Gcloud makes a call to the server to check this,
+   * in future version we can explore the use of '--local-state-only' but that's a relatively new
+   * flag and may not work for all cases.
+   */
+  public boolean hasComponent(SdkComponent component)
+      throws IOException, ExecutionException, CommandExitException {
+    if (!isInstalled()) {
+      return false;
+    }
+    List<String> listComponentCommand =
+        Arrays.asList(
+            getGcloud().toString(),
+            "components",
+            "list",
+            "--format=json",
+            "--filter=id:" + component + " AND state.name:Not Installed");
+
+    String result = new CommandFactory().newCaller(listComponentCommand, null, null).call();
+    return !result.contains(component.toString());
```

Comment:
Using string search instead of parsing the JSON is risky. E.g. imagine at some point in the future there are python and python3 components. A search for python would return true if python3 is installed.